internal/infra/updater.go

```diff
@@ -113,6 +141,127 @@ func NewUpdater(ctx context.Context, cli *client.Client, net *Networks, params *
 	return updater, nil
 }
 
+func createStorageVolumes(hostCfg *container.HostConfig, ctx context.Context, cli *client.Client, net *Networks) (storageContainerID string, volumeNames []string, err error) {
+	log.Printf("Preparing case insensitive filesystem")
+
+	// create container hosting the storage
+	storageContainerCfg := &container.Config{
+		User:  root,
+		Image: storageImageName,
+		Tty:   true, // prevent container from stopping
+	}
+	storageHostCfg := &container.HostConfig{}
+	storageNetCfg := &network.NetworkingConfig{
+		EndpointsConfig: map[string]*network.EndpointSettings{
+			net.noInternetName: {
+				NetworkID: net.NoInternet.ID, // no external access for this container
+			},
+		},
+	}
+	storageContainer, err := cli.ContainerCreate(ctx, storageContainerCfg, storageHostCfg, storageNetCfg, nil, "")
+	if err != nil {
+		err = fmt.Errorf("failed to create storage container: %w", err)
+		return
+	}
+	storageContainerID = storageContainer.ID
+	caseSensitiveVolumeName := "dpdbot-storage-" + storageContainer.ID[:12]
+	caseInsensitiveVolumeName := "dpdbot-nocase-" + storageContainer.ID[:12]
+	volumeNames = []string{caseSensitiveVolumeName, caseInsensitiveVolumeName}
+
+	// start storage container
+	if err = cli.ContainerStart(ctx, storageContainer.ID, container.StartOptions{}); err != nil {
+		tryRemoveStorageVolume(cli, ctx, caseSensitiveVolumeName)
+		tryRemoveStorageVolume(cli, ctx, caseInsensitiveVolumeName)
```

Comment:
Consider deferring an anonymous function to perform the storage volume cleanup when exiting with a non-nill `err`. That way we don't need to remember to do this cleanup on all of the potential error paths.